### ISSUE TITLE
Add a YAML file for creating the dev-environment?

### ISSUE THREAD

**Issue by HTenkanen** [r5py/r5py#19]
@christophfink: Now after starting to play around with this, I notice that we don't yet have a environment file that could be used to easily create a dev-environment for the package. Shall I write one, now as I'm setting things up?

**Comment by HTenkanen** [comment]
We could follow a similar style doing these as many projects, such as [geopandas](https://github.com/geopandas/geopandas/tree/main/ci/envs), i.e. there woud be a folder `ci/envs` and we could add our environments there.

**Comment by christophfink** [comment]
I’m feeling a bit ignorant, but I always thought the requirements.txt/setup.cfg/pyproject.toml already describe all necessary steps? But yes, go ahead. This is (probably?) also important for a matrix testing setup that would be nice (with all the Java stuff coming in, too, and different OS and such)

**Comment by christophfink** [comment]
So environment files are apparently a conda thing, so this should go together with issue #13

**Comment by christophfink** [comment]
Should we close this issue now, and re-open it (or rather new ones), when we need other environments?

**Comment by christophfink** [comment]
Reopening this, we should add [nbstripout](https://anaconda.org/conda-forge/nbstripout) to the developer env.

**Comment by HTenkanen** [comment]
@christophfink Will also add OpenJDK here as a dependency so when developing, we can be sure that our environments share the same version of JDK: https://anaconda.org/conda-forge/openjdk

**Comment by HTenkanen** [comment]
Using conda to install JDK is also extremely convenient so when we instruct how to install r5py it's basically just a couple of steps, 1) install openjdk and 2) install r5py.

**Comment by christophfink** [comment]
perfect!

**Comment by christophfink** [comment]
The GitHub runners only allow certain flavours of JDKs, but the default one seems to work well.

**Comment by christophfink** [comment]
Also make sure these are available:
```
ImportError: The 'folium', 'matplotlib' and 'mapclassify' packages are required for 'explore()'. You can install them using 'conda install -c conda-forge folium matplotlib mapclassify' or 'pip install folium matplotlib mapclassify'.
```

**Comment by HTenkanen** [comment]
Ah right, will add them.

**Comment by christophfink** [comment]
Thanks!

**Comment by HTenkanen** [comment]
> Let’s also make sure these are available:
> 
> ```
> ImportError: The 'folium', 'matplotlib' and 'mapclassify' packages are required for 'explore()'. You can install them using 'conda install -c conda-forge folium matplotlib mapclassify' or 'pip install folium matplotlib mapclassify'.
> ```

Hmm, have you created your environment with conda or did you use something else? In my case, these packages were actually installed together with geopandas. I will still add these to the dev-environment, but just want to understand whether there are actually differences between how conda behaves. 🤔

**Comment by christophfink** [comment]
No, my geopandas comes from PyPi (or actually from pacman, but it’s built using the wheel from PyPi). Might well be that they are pulled in by default using conda

**Comment by HTenkanen** [comment]
> No, my geopandas comes from PyPi (or actually from pacman, but it’s built using the wheel from PyPi). Might well be that they are pulled in by default using conda

Okay, this is what I expected. Then all good. With conda, the environments should be identical, hence I started thinking whether there is something weird going on.
